packages/node_modules/@webex/plugin-meetings/src/reconnection-manager/index.js

```diff
@@ -143,59 +153,29 @@ export default class ReconnectionManager {
 
   /**
    * @param {Meeting} meeting
-   * @param {boolean} [recreate] default false to re attach media if true
    * @returns {Promise}
    * @private
    * @memberof ReconnectionManager
    */
-  reconnectMedia(meeting, recreate = false) {
-    let promise = null;
+  reconnectMedia(meeting) {
+    this.tryCount += 1;
+    LoggerProxy.logger.log(`ReconnectionManager:index#media --> Begin reestablishment of media at ${new Date()}`);
 
-    if (recreate) {
-      promise = Media.attachMedia(meeting.mediaProperties, {
-        meetingId: meeting.id,
-        remoteQualityLevel: meeting.mediaProperties.remoteQualityLevel
-      })
-        .then((peerConnection) => meeting.setRemoteStream(peerConnection));
-    }
-    else {
-      promise = Media.reconnectMedia(meeting.mediaProperties.peerConnection, meeting);
-    }
+    ReconnectionManager.setupPeerconnection(meeting);
 
-    return Promise.resolve(promise)
+    return Media.attachMedia(meeting.mediaProperties, {
+      meetingId: meeting.id,
+      remoteQualityLevel: meeting.mediaProperties.remoteQualityLevel
+    })
+      .then((peerConnection) => meeting.setRemoteStream(peerConnection))
       .then(() => meeting.roap
         .sendRoapMediaRequest({
           sdp: meeting.mediaProperties.peerConnection.sdp,
           roapSeq: meeting.roapSeq,
-          meeting // or can pass meeting ID
-        }));
-  }
-
-  /**
-   * @param {Meeting} meeting
-   * @returns {Promise} see #Meeting.addMedia
-   * @private
-   * @memberof ReconnectionManager
-   */
-  media(meeting) {
-    this.tryCount += 1;
-    LoggerProxy.logger.log(`ReconnectionManager:index#media --> Begin reestablishment of media at ${new Date()}`);
-
-
-    // TODO: chekc peerconnection status
-    // create an offer
-    return this.reconnectMedia(meeting)
-      .then((media) => Promise.resolve(media))
-      .catch((err) => {
+          meeting, // or can pass meeting ID
+          reconnect: true // Need to check if its a reconnect after rejoin or media inactivity
+        })).catch((err) => {
```

Comment:
It looks like the `execute` function relies on this failure to do the retry logic. We're only catching it there though, not elevating the error as a promise rejection on `reconnectMedia`.

Reply:
Agree, we should throw from here

Reply:
done